src/DssDirectDepositAaveDai.t.sol

```diff
@@ -505,7 +561,260 @@ contract DssDirectDepositAaveDaiTest is DSTest {
+        uint256 amountToBorrow = currentLiquidity + amountSupplied / 2;
+        pool.borrow(address(dai), amountToBorrow, 2, 0, address(this));
+
+        deposit.cage();
+
+        hevm.warp(block.timestamp + deposit.tau());
+
+        uint256 daiEarned = adai.balanceOf(address(deposit)) - pink;
+
+        VowAbstract(vow).heal(
+            min(
+                vat.sin(vow) - VowAbstract(vow).Sin() - VowAbstract(vow).Ash(),
+                vat.dai(vow)
+            )
+        );
+        uint256 originalSin = vat.sin(vow);
+        uint256 originalDai = vat.dai(vow);
+        // If the whole Sin queue would be cleant by someone,
+        // originalSin should be 0 as there is more profit than debt registered
+        assertGt(originalDai, originalSin);
+        assertGt(originalSin, 0);
+
+        deposit.cull();
+
+        // After cull, the debt of the position is converted to bad debt
+        assertEq(vat.sin(vow), originalSin + part * RAY);
+
+        // CDP grabbed and ink moved as free collateral to the deposit contract
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(deposit));
+        assertEq(ink, 0);
+        assertEq(art, 0);
+        assertEq(vat.gem(ilk, address(deposit)), pink);
+        assertGe(adai.balanceOf(address(deposit)), pink);
+
+        // MCD shutdowns
+        end.cage();
+        end.cage(ilk);
+
+        assertEq(vat.sin(vow), originalSin + part * RAY - originalDai);
+
+        deposit.uncull();
+        VowAbstract(vow).heal(min(vat.sin(vow), vat.dai(vow)));
+
+        // So the position is restablished
+        (ink, art) = vat.urns(ilk, address(deposit));
+        assertEq(ink, pink);
+        assertEq(art, part);
+        assertEq(vat.gem(ilk, address(deposit)), 0);
+        assertGe(adai.balanceOf(address(deposit)), pink);
+        assertEq(vat.sin(vow), 0);
+
+        // Call skim manually (will be done through deposit anyway)
+        // Position is again taken but this time the collateral goes to the End module
+        end.skim(ilk, address(deposit));
+        VowAbstract(vow).heal(min(vat.sin(vow), vat.dai(vow)));
+
+        (ink, art) = vat.urns(ilk, address(deposit));
+        assertEq(ink, 0);
+        assertEq(art, 0);
+        assertEq(vat.gem(ilk, address(deposit)), 0);
+        assertEq(vat.gem(ilk, address(end)), pink);
+        assertGe(adai.balanceOf(address(deposit)), pink);
+        assertEq(vat.sin(vow), originalSin + part * RAY - originalDai);
+
+        // We try to unwind what is possible
+        deposit.exec();
+        VowAbstract(vow).heal(min(vat.sin(vow), vat.dai(vow)));
+
+        // A part can't be unwind yet
+        assertEq(vat.gem(ilk, address(end)), amountSupplied / 2);
+        assertGt(adai.balanceOf(address(deposit)), amountSupplied / 2);
+        assertLe(vat.sin(vow), originalSin + part * RAY - originalDai - (amountSupplied / 2) * RAY);
+        assertGe(vat.sin(vow), originalSin + part * RAY - originalDai - (amountSupplied / 2 + 1) * RAY);
```

Comment:
├╴call 0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B::sin(@0xA950524441892A31ebddF91d3cEEFa04Bf454466) (src/DssDirectDepositAaveDai.t.sol:779)
      │  └╴← (0)
      ├╴log("Error: a >= b not satisfied [uint]") (lib/ds-test/src/test.sol:224)
      ├╴log_named_uint("  Value a", 0) (lib/ds-test/src/test.sol:225)
      ├╴log_named_uint("  Value b", 115792089237316195423570961774909335961466340321992856468664914230636615358125) (lib/ds-test/src/test.sol:226)

Reply:
I think this should be fixed now using directly the `assertEqApprox` function.